easypermissions/src/main/java/pub/devrel/easypermissions/EasyPermissions.java

```diff
@@ -85,205 +80,220 @@ public static boolean hasPermissions(@NonNull Context context, @NonNull String..
-            @NonNull String rationale, @StringRes int positiveButton, @StringRes int negativeButton,
-            final int requestCode, @NonNull final String... perms) {
-
-        RationaleDialogFragment fragment = RationaleDialogFragment
-                .newInstance(positiveButton, negativeButton, rationale, requestCode, perms);
-        fragment.show(fragmentManager, DIALOG_TAG);
+    public static void requestPermissions(@NonNull Fragment fragment,
+                                          @NonNull String rationale,
+                                          int requestCode,
+                                          @NonNull String... perms) {
+        requestPermissions(
+                fragment,
+                rationale,
+                android.R.string.ok,
+                android.R.string.cancel,
+                requestCode,
+                perms);
     }
 
     /**
-     * Show an {@link AlertDialog} explaining permission request rationale.
+     * Request a set of permissions, showing rationale if the system requests it.
+     *
+     * @param fragment       Fragment requesting permissions. Should implement {@link
+     *                       ActivityCompat.OnRequestPermissionsResultCallback} {@code
+     *                       android.support.v13.app.FragmentCompat.OnRequestPermissionsResultCallback}
+     * @param rationale      a message explaining why the application needs this set of permissions,
+     *                       will be displayed if the user rejects the request the first time.
+     * @param positiveButton custom text for positive button
+     * @param negativeButton custom text for negative button
+     * @param requestCode    request code to track this request, must be < 256.
+     * @param perms          a set of permissions to be requested.
+     * @see android.Manifest.permission
      */
-    private static void showRationaleAlertDialog(
-            @NonNull final Object object, @NonNull String rationale,
-            @StringRes int positiveButton, @StringRes int negativeButton,
-            final int requestCode, @NonNull final String... perms) {
-
-        Activity activity = getActivity(object);
-        if (activity == null) {
-            throw new IllegalStateException("Can't show rationale dialog for null Activity");
+    @SuppressWarnings("NewApi")
+    public static void requestPermissions(@NonNull Fragment fragment,
+                                          @NonNull String rationale,
+                                          @StringRes int positiveButton,
+                                          @StringRes int negativeButton,
+                                          int requestCode,
+                                          @NonNull String... perms) {
+        if (hasPermissions(fragment.getContext(), perms)) {
+            notifyAlreadyHasPermissions(fragment, requestCode, perms);
+            return;
         }
 
-        new AlertDialog.Builder(activity)
```

Comment:
This was used for `api level < something lower than M` which means it's useless because we already have the permissions if we are lower than M.